### ISSUE TITLE
[Bug]: JAMD disappearing when RFTools Dimensions are created

### ISSUE THREAD

**Issue by Dreavers** [FTBTeam/FTB-Modpack-Issues#1075]
### Mod

JAMD Just Another Mining Dimension

### Mod version

1.6.1

### Forge / Fabric version

Forge Version 40.1.52

### Modpack & version

FTB Presents Direwolf 20 1.18 v1.6.1

### What issue are you having?

jamd: Just Another Mining Dimension is conflicting with RFTools Dimensions. If I create a RFT Dimension and reload the game the mining dimension disappears. using Digit Dimlets doesn't help the issue. Deleting all the RFT Dimension folders from my game allows me to load the game and enter the mining dimension, but closing the game and opening it with any RFT dimensions causes the jamd dimension to go missing again

### Steps to reproduce

1. Create a JAMD Mining Dimension and place a waystone
2. Create an RFTools Dimension, with or without a digit dimlet
3. Power the RFTools Dimension
4. Port to it
5. Return to the overworld and save and exit the game
6. Start the game
7. Try to teleport to the JAMD dimension using either the JAMD block or the waystone

### Anything else to note?

Deleting all your RFT Dimension folders fixes it temporarily until you start the game again. I suspect somehow RFT Dimensions is starting with the same dimension number JAMD is located at but it's hard for me to tell.

**Comment by sadovsf** [comment]
Same goes with Just another void dimension.

**Comment by sadovsf** [comment]
For me it was enough to first time create dimension and get into it few times. Inside i did use Flux networks for powering teleporter and walked for few minutes close to spawn. Based on backups it broke right around time dimension was spawned. It was properly placed in its own folder as you claim but for some reason mc did not see javd and jamd at all until i deleted that created dimension from data and restarted server. What could help is, that first i did create and use Just another dimensions and only then started to mess with RFTools one could be order.

Here is link to backup where it is broken. 
https://www.dropbox.com/s/zp55uqzujmkx31g/broken_DW20_18_save_backup.zip?dl=0

Setup used and everything needed is at `overworld` at coords `410 57 1844`. Void portal is then close by at `410 57 1881`. It was played on dedicated server although it was broken even when same world loaded in single later.

**Comment by sadovsf** [comment]
Let me know when you no longer need it so i can delete it (i dont have much space there)

**Comment by TheonlyTazz** [comment]
downloaded thanks, ill have a look later today

**Comment by sadovsf** [comment]
Did it help you? I could repro it probably again if needed

**Comment by TheonlyTazz** [comment]
i am going to forward this to RFtools Issues tracker and hope i can fix this asap

**Comment by TheonlyTazz** [comment]
> Did it help you? I could repro it probably again if needed

after trying it multiple times on your save file, it is correct that the void dimension is inaccessable, but after removing/renaming the RFTools dimension folder, the JAVD dimension is accesable, and even after creating a new dimension, the JAVD dimension stays available, so i could not 100% replicate this. I also have no actual log for this, which McJty would like to have...
maybe you @sadovsf could provide them if possible

**Comment by sadovsf** [comment]
There is some where i was trying to get into dimension when it was broken
[2022-07-10-2.log.gz](https://github.com/FTBTeam/FTB-Modpack-Issues/files/9119362/2022-07-10-2.log.gz)

**Comment by Raidobw2** [comment]
Hi @sadovsf, I've been able to take a look at your world with the help of Tazz, and I believe the error with those dimensions was caused by a failure to save `level.dat`, an important level file that holds dimension data. Possibly:
- A power cut happened while saving
- The `level.dat` file was locked by an antivirus for a second
- The `level.dat` file was locked by something else

I'd like to tell you that the solution to this problem would be to restore the `level.dat` file from an earlier backup that contained dimension data. Currently, yours doesn't have any, therefore I think of it as corrupted. There is no mod update that will solve the problem you currently are experiencing, simply there is missing data. You could use NBTExplorer to open `level.dat` files and see if there is a "WorldGenSettings" key with any information in it.

Technical explanation is said in the linked issue to McJty's tracker above if you would like to read it, please let me know if you have any questions so we can help you recover world data properly.

Additionally, you could restore RFTools Dimension data in your world through the same method. Take the copies of those files from the same day your `level.dat`  contains the correct dimension data and replace them.
They are in `world/data` and called `RFToolsDimensions.dat` and `TPDestinations.dat` respectively. Note that doing so may render dimensions added in the past days inexistent, as if you had restored a backup. Simply be aware of this when choosing to do it.

**Comment by sadovsf** [comment]
Thank you for the info. Iam not aware of any issue around this I play on DS where only mc server is running. It would be hell of a coincidence that this happend after roughly 60h of play time right when I started messing with dimensions and when same happened to original reporter. But still not impossible I guess. World I created was full of honey blocks and was really expensive to render which should be an issue on client while saving happens on server so not sure that was an issue. Important part is, that it should not be happening and I will simply continue with dimensions and let you know in case it breaks again.

I managed to restore everything already next day. But thank you for guide. What I did was to delete all rf tools dimensions and rebooted. That was enough to restore it to work actually.
